gdshelpers/parts/waveguide.py

```diff
@@ -409,17 +409,28 @@ def add_route_single_circle_to(self, final_coordinates, final_angle, max_bend_st
         d = abs(radius * np.tan(diff_angle / 2))
 
         tmp_wg = Waveguide.make_at_port(self._current_port)
-        tmp_wg.add_straight_segment(distance[0] - d)
+        tmp_wg.add_straight_segment(length=distance[0] - d)
         tmp_wg.add_bend(-diff_angle, radius)
+        if not on_line_only:
+            tmp_wg.add_straight_segment(distance[1] - d)
+        if final_width is not None:
+            tmp_segment_lengths = [segment[2] for segment in tmp_wg.get_segments()]
```

Comment:
Creating a waveguide for just calculating the length is rather expensive - why not directly calculate it (it's just two straight segments and a bend)

Reply:
Added.